test/test_unary_ufuncs.py

```diff
@@ -378,6 +397,61 @@ def test_batch_vs_slicing(self, device, dtype, op):
 
         self.assertEqual(actual, expected)
 
+    def _compare_out_variant_based_on_device(self, op, input, out):
+        out_dtype = out.dtype
+        if self.device_type == 'cuda':
+            # As input are dynamically casted
+            # in kernel, cast input to out_dtype
+            expected = op(input.to(out_dtype))
```

Comment:
This is not generally true. For example, op(fp16, out=fp64) != op(fp64). The former computation occurs in float16, while the latter computation occurs in the wider, higher precision float64.

Reply:
Explanation below.